sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -76,7 +81,23 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
           createdAt: currentDateTime,
         } as StudentAppealRequest),
     );
-    return this.repo.save(studentAppeal);
+
+    // Update the student appeal and files if exist using the same entity manager
+    // so that they are processed within the same transaction
+    return await this.dataSource.transaction(async (entityManager) => {
+      const student = await this.studentService.getStudentByUserId(userId);
+      const uniqueFileNames: string[] = studentAppealRequests.flatMap(
+        (studentAppeal) => studentAppeal.files,
+      );
+      this.studentFileService.updateStudentFiles(
+        student.id,
```

Comment:
Student id can be sent from the controller through the token. Please use the `StudentUserToken` in the controller.

![image](https://github.com/bcgov/SIMS/assets/54600590/6ef275c8-20c3-44a1-8e82-273da3f1eaad)